### DISCUSSION TITLE
Ideas for VBA coding

### DISCUSSION THREAD

**Discussion by PW25** [SAP-samples/integrated-business-planning-excel-addin-vba-samples#5] [Ideas]
Hi all,

in case you have new ideas for VBA coding, enhancements or use cases, please post them right here.

**Comment by vvdor** [comment]
Possible idea: draw separator-lines between a group of key figures, i.e. whenever the product ID changes, or Customer ID changes, etc. Like that, the data in large views is visually separated and is easy to read.

**Reply by PW25** [reply]
Thanks for your feedback. Let me follow-up on that topic.

**Reply by PW25** [reply]
We would like to provide this feature native using the IBP formatting sheet. I added it to our backlog.

**Reply by ayanbishnu** [reply]
@PW25 thanks a lot for adding it int he backlog. For the meantime i have coded in VBA and it is working fine. only downside for the development is user have to refresh the planning view as i have provided them with a activex checkbox, where if checked the custom format will be executed and if unchecked standard format is displayed.

I have attached the excel macro templates and also the modules has been saved separately as .BAS file.

[Redbull.zip](https://github.com/SAP-samples/integrated-business-planning-excel-addin-vba-samples/files/7985401/Redbull.zip)

**Comment by ayanbishnu** [comment]
We have a formatting enhancement for our planning views. We would like to activate the option "repeat row headers" but in a way that only the first row is shown. All repeated rows should have the same text color as the background.
Please refer to the attachment. The headers should be repeated only in the yellow rows (not in the ones between as well). Please help and advice.
![Sample](https://user-images.githubusercontent.com/96914583/147825847-4bbf7cd4-9d87-42ec-86c9-0e0f6adc7380.png)

**Reply by PW25** [reply]
We would like to provide this feature native using the IBP formatting sheet. I added it to our backlog.

**Reply by ayanbishnu** [reply]
@PW25 thanks a lot for adding it int he backlog. For the meantime i have coded in VBA and it is working fine. only downside for the development is user have to refresh the planning view as i have provided them with a activex checkbox, where if checked the custom format will be executed and if unchecked standard format is displayed.

I have attached the excel macro templates and also the modules has been saved separately as .BAS file.

[Redbull.zip](https://github.com/SAP-samples/integrated-business-planning-excel-addin-vba-samples/files/7985426/Redbull.zip)

**Comment by Vincent-Mia-Edie-Verheyen** [comment]
@PW25 @JVotteler Would it be possible to provide a snippet of how it would be possible to unallow the removal of a filter from a certain Planning View Template's worksheet (after having opened it via New View > From Template ...). 

Perhaps code-inspiration could be found in the IBP web UI Global Configuration parameter "FORCE_PLANNING_VIEW_FILTER" (Description: _Force user to set up an attribute-based filter in planning views_)?

The goal is, that one can disallow the removal of a Planning Filter's attribute, for a certain Planning View's sheet only. The default Planning Filter that comes with a Planning View Template, for example, might be called PRODUCT GROUP XYZ, and might be defined as Product Group = XYZ. The requirement for business now is that users (when loading a Planning View from the template) are not able to remove this filter (of Product Group).

**Reply by Vincent-Mia-Edie-Verheyen** [reply]
Please see https://github.com/SAP-samples/integrated-business-planning-excel-addin-vba-samples/discussions/13

**Reply by PW25** [reply]
Hi @Verheyen-Vincent. Sorry for the huge delay. At the moment there's no ideal option to achieve this scenario. We only have the filter options defined in https://help.sap.com/docs/SAP_INTEGRATED_BUSINESS_PLANNING/b28ffdd739bf45678ef36c44e64652d7/fb0633785cc94fc8a3776827e6338df0.html?locale=en-US. Improving this feature is on our list, but not yet assigned to a release. Ideally you can define during the template creation what attributes and values shouldn't be removed from the planning view.

**Comment by Vincent-Mia-Edie-Verheyen** [comment]
@PW25 + Another question I have would be on how, best-practice, to throw an error message when, in a certain Planning View, there is 1 of a certain amount of pre-defined attributes (the base planning level of a certain Key Figure) not present in the Planning View.

https://github.com/SAP-samples/integrated-business-planning-excel-addin-vba-samples/blob/main/VBA%20samples/SAP%20IBP%20Hooks/Sample%20code%20for%20SAP%20IBP%20Hooks.pdf already shows a code for 'If the opened worksheet includes the master data type "Product"

However, this is for Master Data Workbooks. My question is how to do the same for a Planning View.

**Reply by Vincent-Mia-Edie-Verheyen** [reply]
I have created a seperate for discussion for this topic here: https://github.com/SAP-samples/integrated-business-planning-excel-addin-vba-samples/discussions/12
